File: task-3/Go/Sub-Task 4/main.go
package main

import (
	"fmt"
	"strings"
	"io"
	"os"
    "bufio"
    "strconv"
)

func main() {
	fi, err := os.Open("input.txt")
    if err != nil {
        panic(err)
    }
    defer fi.Close()
    
    scanner := bufio.NewScanner(fi)
    scanner.Scan()

    if err := scanner.Err();
    err != nil {
        panic(err)
        return
    }

    n, err := strconv.Atoi(strings.TrimSpace(scanner.Text()))
    if err != nil {
        panic(err)
    }
		
	var counter int = n%2
	var seq string = ""

	for i := 0; i< n; i++{
		if counter == 0 {
			counter += 2
			continue
		}
		
		stars := strings.Repeat("*",counter)
		spaces := strings.Repeat(" ", int(((n-counter)/2)))
		
		if i < int(n/2) {
			counter += 2
		} else {
			counter -= 2
		}
			
		seq += spaces+stars+spaces+"\n"
	}
	
	fo, err := os.Create("output.txt")
    if err != nil {
        panic(err)
    }
    defer fo.Close()
	
	_, err = io.WriteString(fo, seq)
    if err != nil {
        fmt.Println("Error writing to file:", err)
        return
    }
}

File: task-3/Java/Sub-Task 4/Main.java

import java.io.BufferedWriter;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class Main
{
    public static String repeat(String s, int n)
    {
        return new String(new char[n]).replace("\0", s);
    }
    
	public static void main(String[] args)
	{
	    int n = 0;
	    String pattern = "";
	    
	    try (BufferedReader br = new BufferedReader(new FileReader("input.txt"))) {
            n = Integer.parseInt(br.readLine());
        } catch (IOException e) {
            e.printStackTrace();
        }
        
        int counter = n%2;
        
        for (int i = 0; i < n; i++)
		{
        	if(counter == 0)
        	{
        		counter += 2;
        		continue;
        	}
        	
        	String stars = repeat("*", counter);
        	String spaces = repeat(" ", (int) (n-counter)/2);
        
        	if(i < (int) n/2) {
        		counter += 2;
        	} else {
        		counter -= 2;
        	}
        	
        	pattern += spaces+stars+spaces+"\n";
        	
        }
        
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("output.txt"))) {
            writer.write(pattern);
            writer.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
	}
}